src/spox/_attributes.py

```diff
@@ -30,6 +32,12 @@
 AttrIterableT = TypeVar("AttrIterableT", bound="_AttrIterable")
 
 
+class AttributeTypeError(TypeError):
```

Comment:
Perhaps nitpick: Is it useful to disambiguate the specific TypeError? There isn’t a stable exception type established anyway.

Reply:
> Is it useful to disambiguate the specific TypeError?

It's useful if we want to cover those specific error cases, e.g. the test:

https://github.com/Quantco/spox/blob/2c3d882156d4186d2f0d1cda99e27ea0feaf2150/tests/test_standard_op.py#L69-L75

In general, it is encouraged to have custom exception types in Python, because it makes exceptions-for-control-flow less error-prone.

Reply:
IMHO, we should stick with standard error types (i.e. `TypeError`) for now. Introducing more fine-grained errors should better be done project-wide rather than piece-meal and on the fly.

Reply:
It's really not a big deal and it's usually done a lot and considered a good practice. Here it extends `TypeError`, and its name suggests it is coupled with onnx attributes. Let's stick with it, it makes exception handling easier.

Reply:
From the user's point of view, all public functions from the opset are regular Python functions with only `spox.Var`, built-in, NumPy types in their signature. The concept of an "Attribute" being constructed behind the scenes is an implementation detail that does not need to be part of the public interface. Conceptually, writing `op.constant(value_ints=["a"])` is exactly the same error as writing `range("foo")`.